Add wifi provisioning BLE service

wentong-li opened the pull request:
Extract common part from wifi provisioning sample, and form it as a BLE service.

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10178/3/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10178/3/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| desktop52_verification | X | | |
| test-fw-nrfconnect-ble | X | | |
| test-fw-nrfconnect-ble_samples | X | | |
| test-fw-nrfconnect-zigbee | X | | |
| test-sdk-find-my | X | | |
| test-sdk-homekit | X | | |
| test-sdk-wifi | X | | |
| test-sdk-wifi-radio | X | | |
| test-sdk-wifi-scan | X | | |
| test-sdk-wifi-sta | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-10178/3/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

wentong-li commented:
Hi @richabp . The doc is reorganized as follows:

`readme.rst`: previously it is a navigation page containing two subpages. Now it holds the contents of `sample_description.rst` and no longer has subpage. **Now it is for the sample.**
`sample_description.rst`: removed
`implementaion_details.rst`: it is moved to doc/nrf/libraries/bluetooth_services/services, and renamed as `wifi_prov.rst`. **It is for the service.**

Hope my explanation can help.

kittydepa commented on doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst:
```diff
@@ -1,102 +1,25 @@
-The sample uses four message types:
-
-* ``Info``: Allows the configurator to know the capability of a device before the provisioning process happens.
+This module implements of an Bluetooth GATT service for Wi-Fi provisioning.
```
```suggestion
This module implements a Bluetooth® GATT service for Wi-Fi® provisioning.
```

kittydepa commented on doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst:
```diff
@@ -1,102 +1,25 @@
-Exchanging messages requires a secure transport method.
-The Wi-Fi Provisioning Service sample uses Bluetooth® Low Energy (LE).
-You can assign different security levels to components under one service to fine-tune access control.
+* GATT service inerface: Defines GATT service and provide an interface for configurator to exchange messages.
```
```suggestion
* GATT service inerface: Defines GATT service and provides an interface for the configurator to exchange messages.
```

richabp commented on doc/nrf/libraries/bluetooth_services/services/wifi_prov.rst:
```diff
@@ -155,42 +78,100 @@ The purpose of each characteristic is as follows:
 
 The configuration management component manages Wi-Fi configurations.
-It uses the wifi_credentials library to handle the configurations in flash.
+It uses the :ref:`lib_wifi_credentials` to handle the configurations in flash.
```
```suggestion
It uses the :ref:`lib_wifi_credentials` library to handle the configurations in flash.
```

richabp replied:
```suggestion
It uses the :ref:`lib_wifi_credentials` library to handle the configurations in flash.
```

richabp reviewed (changes requested):
Please add an entry for the updated Provisioning sample in the change log.

kittydepa reviewed (commented):
Added one article, within Richa's suggested edit.

wentong-li commented on samples/wifi/provisioning/README.rst:
```diff
@@ -7,13 +7,133 @@ Wi-Fi: Provisioning Service
+Features
+********
+
+Kconfig options
```
I will add a section called "configuration" to list all kconfig options we provide, and explain their functions

bama-nordic reviewed (approved):
LGTM

kittydepa commented on samples/wifi/provisioning/README.rst:
```diff
@@ -7,13 +7,137 @@ Wi-Fi: Provisioning Service
+Overview
+********
+
+With this sample, you can provision a Wi-Fi® device that lacks input or output capability using the :ref:`wifi_prov_readme` library.
```
```suggestion
With this sample, you can provision a Wi-Fi device that lacks input or output capability using the :ref:`wifi_prov_readme` library.
```

kittydepa commented on samples/wifi/provisioning/README.rst:
```diff
@@ -7,13 +7,137 @@ Wi-Fi: Provisioning Service
+The sample is divided into three parts:
+
+* Task and event handling component: Handles provisioning-related tasks and events.
+* Transport layer: Based on Bluetooth® Low Energy and exchanges information between the device and the nRF Wi-Fi Provisioner app.
```
```suggestion
* Transport layer: Based on Bluetooth Low Energy and exchanges information between the device and the nRF Wi-Fi Provisioner app.
```

kittydepa reviewed (changes requested):
Two small changes, to remove trademark symbols since they have already been used in the first instance.
